packages/byods/src/data-source-service/index.ts

```diff
@@ -0,0 +1,133 @@
+      }
+    } catch (error) {
+      console.log('Encountered some error, error is', error);
+    }
+  }
+
+  /**
+   * This Private method will start auto refreshing the DataSource token with interval as tokenLifetimeMinutes.
+   * @param {string} dataSourceId The id of data source
+   * @param {number} tokenLifetimeMinutes The Life time minutes for the data source. <=1440
+   * @returns {Promise<void>}
+   */
+
+  private async startAutoRefresh(
+    dataSourceId: string,
+    tokenLifetimeMinutes?: number
+  ): Promise<void> {
+    try {
+      // Call the get method to fetch the dataSource details & then call the update method to update the token!
+      const payloadForDataSourceUpdateMethod: any = {};
+      await this.dataSourceClient
+        .get(dataSourceId)
+        .then((response: any) => {
+          const jwsToken: any = response.jwsToken;
+          const jwsTokenPayload: any = this.decodeJWSTokenAndGetPayload(jwsToken);
+          payloadForDataSourceUpdateMethod.schemaId = response.schemaId;
+          payloadForDataSourceUpdateMethod.url = jwsTokenPayload.payload.com.cisco.datasource.url;
+          payloadForDataSourceUpdateMethod.subject = jwsTokenPayload.payload.sub;
+          payloadForDataSourceUpdateMethod.audience = jwsTokenPayload.payload.aud;
+          payloadForDataSourceUpdateMethod.nonce = uuidv4();
+          payloadForDataSourceUpdateMethod.tokenLifetimeMinutes = tokenLifetimeMinutes;
+        })
+        .catch((error: any) => {
+          console.log('error while calling the get method to update the dataSource', error);
+        });
+
+      await this.dataSourceClient
+        .update(dataSourceId, payloadForDataSourceUpdateMethod)
+        .then((response) => {
+          console.log('dataSource has been updated successfully!', response);
+        })
+        .catch((error) => {
+          console.log('error while updating the dataSource', error);
+        });
+      console.log('dataSource has been refreshed successfully');
+    } catch (error) {
+      console.log('Got error while starting auto refreshing dataSource token', error);
+    }
+  }
+
+  /**
+   * This method will stop auto refreshing the DataSource token.
+   * @returns {void}
+   */
+
+  public stopAutoRefresh(): void {
+    if (this.timer) {
+      clearInterval(this.timer);
+      console.log('timer has been cleared successfully.');
+    } else {
+      console.info('timer has not started yet!');
+    }
+  }
+
+  /**
+   * This Private method decodes the JWS token that accepts the parameter which is a token.
+   * @param {string} token A JWS token
+   * @returns {Promise<JWTPayload>} This method return a promise which eventually resolved to decoded payload.
+   */
+
+  // Function to decode a JWS token
+  private async decodeJWSTokenAndGetPayload(token: string): Promise<{payload: JWTPayload} | null> {
+    try {
+      // Decode the token
+      const decoded = decodeJwt(token);
```

Comment:
Doesn't this require an await?

Reply:
**_decodeJwt is not an async function, so it should work without await keyword._**